app/models/booking.rb

```diff
@@ -1,4 +1,8 @@
 class Booking < ApplicationRecord
   belongs_to :user
   belongs_to :piece
+
+  validates :start_on, :end_on, :user, :piece, presence: true
+  validates :end_on, presence: true
+  validates :status, default: "available"
```

Comment:
Je pense que le statu "available" est a rattaché à l'oeuvre s'il n'y a pas de réservation en attente. Il s'agirait donc d'un statut conditionnel :

1.  Il y a une réservation en cours => "Not available" => Return date = ...
2. Il n'y a pas de réservation en cours => Available

=> Le status pourrait donc dépendre d'une plage de date.